### PR TITLE
Add extra assertions to generate features tests 

### PR DESCRIPTION
**Pull request by turkeylurkey**
Also adjust other tests not to interfere with each other.

### CODE REVIEW

File: src/test/groovy/io/openliberty/tools/gradle/DevTest.groovy [hunk 1]
```diff
@@ -181,14 +180,11 @@ class DevTest extends BaseDevTest {
...
     public void generateFeatureTest() throws Exception {
-        final String GENERATED_FEATURES_FILE_NAME = "generated-features.xml";
-        final String SERVER_XML_COMMENT = "Plugin has generated Liberty features"; // the explanation added to server.xml
-        final String NEW_FILE_INFO_MESSAGE = "This file was generated by the Liberty Gradle Plugin and will be overwritten"; // the explanation added to the generated features file
-
+        assertFalse(verifyLogMessage(10000, "batch-1.0", errFile)); // not present on server yet
         // Verify generate features runs when dev mode first starts
         assertTrue(verifyLogMessage(10000, RUNNING_GENERATE_FEATURES));
-        int generateFeaturesCount = countOccurrences(RUNNING_GENERATE_FEATURES, logFile);
-        assertFalse(verifyLogMessage(10000, "batch-1.0", errFile)); // not present on server yet
```
Comment (kathrynkodama) [inline src/test/groovy/io/openliberty/tools/gradle/DevTest.groovy:191]: Why did we remove this line? Should we remove in ci.maven as well? https://github.com/OpenLiberty/ci.maven/pull/1540/files#diff-94179a656c021421e9e696ddf72cf24f58d8247ef199afdfdab55895d18c0f90R267

File: src/test/groovy/io/openliberty/tools/gradle/DevTest.groovy [hunk 2]
```diff
@@ -222,31 +218,36 @@ class DevTest extends BaseDevTest {
...
         // Need valid code for testing
-        String goodCompile = "Source compilation was successful.";
-        int goodCount = countOccurrences(goodCompile, logFile);
+        int goodCount = countOccurrences(COMPILATION_SUCCESSFUL, logFile);
+        int regenerateCount = countOccurrences(REGENERATE_FEATURES, logFile);
         replaceString(badCode, goodCode, helloBatchSrc);
-        assertTrue(verifyLogMessage(10000, goodCompile, goodCount+1));
+        assertTrue(verifyLogMessage(10000, COMPILATION_SUCCESSFUL, goodCount+1));
+        // after successful compilation run generate features. "Regenerate" message should appear after.
+        // Restore these tests once issue 757 is fixed.
```
Comment (kathrynkodama) [inline src/test/groovy/io/openliberty/tools/gradle/DevTest.groovy:247]: Suggestion to mark as TODO so that it shows up in problems views of IDEs

**Review by kathrynkodama — COMMENTED** [review]
Should we tag the start and end of tests similar to how it is done in the corresponding ci.maven PR?

File: src/test/groovy/io/openliberty/tools/gradle/DevTest.groovy (hunk 1 above)
Comment (turkeylurkey) [reply src/test/groovy/io/openliberty/tools/gradle/DevTest.groovy:191]: It just moved up to line 183 in the new file.

File: src/test/groovy/io/openliberty/tools/gradle/DevTest.groovy (hunk 2 above)
Comment (turkeylurkey) [reply src/test/groovy/io/openliberty/tools/gradle/DevTest.groovy:247]: Done and moved to file DevRecompileTest.java

File: src/test/groovy/io/openliberty/tools/gradle/DevTest.groovy [hunk 3]
```diff
@@ -54,16 +54,16 @@ class DevTest extends BaseDevTest {
...
 
-        replaceString("</feature>", "</feature>\n" + "    <feature>mpHealth-2.0</feature>", srcServerXML);
+        replaceString("</feature>", "</feature>\n" + "    <feature>mpFaultTolerance-2.0</feature>", srcServerXML);
 
         // check that features have been generated
         assertTrue(verifyLogMessage(10000, RUNNING_GENERATE_FEATURES, ++generateFeaturesCount)); // task ran
 
         // check for server configuration was successfully updated message in messages.log
         File messagesLogFile = new File(targetDir, "wlp/usr/servers/defaultServer/logs/messages.log");
         assertTrue(verifyLogMessage(60000, "CWWKG0017I", messagesLogFile));
```
[the remark this answers is not in the document]
Comment (turkeylurkey) [reply src/test/groovy/io/openliberty/tools/gradle/DevTest.groovy:64]: I added all four server messages from ci.maven even though one is not used yet.

**Pull request closed without merging**